src/test/java/org/lookout_studios/meals_management_system/meals_management_system/RegistrationServiceTests.java

```diff
@@ -0,0 +1,114 @@
+            @Override
+            public boolean hasError(ClientHttpResponse response) throws IOException {
+                return false;
+            }
+
+            @Override
+            public void handleError(ClientHttpResponse response) throws IOException {
+            }
+        });
+    }
+
+    @AfterAll
+    public static void afterAll() {
+        wireMockServer.stop();
+    }
+
+    @AfterEach
+    public void afterEach() {
+        wireMockServer.resetAll();
+    }
+
+    private String requestUrl = "http://localhost:8080/register";
+
+    /**
+     * Ensures that /register requests with invalid emails generate
+     * Bad Request Response
+     * 
+     * @throws Exception
+     */
+    @Test
+    public void registerInvalidEmailUserFail(WireMockRuntimeInfo wMockRuntimeInfo) throws Exception {
+        JSONObject requestBody = new JSONObject();
+        requestBody.put("email", "abc-@mail.com");
+        requestBody.put("password", "12345678");
+        String responseBody = new ResponseBody(HttpStatus.BAD_REQUEST).getResponseBody();
+        WireMock.stubFor(
+                WireMock.post(WireMock.urlEqualTo("/register"))
+                        .willReturn(WireMock.aResponse().withStatus(400).withBody(responseBody)));
+        ResponseEntity<String> response = restTemplate.postForEntity(requestUrl, requestBody, String.class);
+        assertEquals(HttpStatus.BAD_REQUEST.value(), response.getStatusCode().value());
+    }
+
+    /**
+     * Ensures that /register requests with invalid passwords generate
+     * Bad Request response
+     *
+     * @throws Exception
+     */
+    @Test
+    public void registerInvalidPasswordUserFail() throws Exception {
+        JSONObject requestBody = new JSONObject();
+        requestBody.put("email", "abc@mail.com");
+        requestBody.put("password", null);
+        String responseBody = new ResponseBody(HttpStatus.BAD_REQUEST).getResponseBody();
+        WireMock.stubFor(
+                WireMock.post(WireMock.urlEqualTo("/register"))
+                        .willReturn(WireMock.aResponse().withStatus(400).withBody(responseBody)));
+        ResponseEntity<String> response = restTemplate.postForEntity(requestUrl, requestBody, String.class);
+        assertEquals(HttpStatus.BAD_REQUEST.value(), response.getStatusCode().value());
+
+    }
+
+    /**
+     * Ensures that valid /register requests generate OK response
+     *
+     * @throws Exception
+     */
+    @Test
+    public void registerExistingUserFail() throws Exception {
+        JSONObject requestBody = new JSONObject();
+        requestBody.put("email", "abc@mail.com");
+        requestBody.put("password", "12345678");
+        String responseBody = new ResponseBody(HttpStatus.OK).getResponseBody();
+        WireMock.stubFor(
+                WireMock.post(WireMock.urlEqualTo("/register"))
+                        .willReturn(WireMock.aResponse().withStatus(200).withBody(responseBody)));
```

Comment:
Magic value (200).

Reply:
Refer to [this comment](https://github.com/Mateusz-Dobrzynski/meals-management-system/pull/9#discussion_r1300297585)